docs/sec-blueprint/07-Hardening.md

```diff
@@ -0,0 +1,597 @@
+```
+
+  These kernel debug symbols are enabled by other config items in the
+  kernel. Care should be taken to disable those also. If
+  CONFIG\_DEBUG\_INFO cannot be disabled then enabling
+  CONFIG\_DEBUG\_INFO\_REDUCED is second best.
+
+### Disable access to a kernel core dump
+
+  This kernel configuration disables access to a kernel core dump from
+  user space -- if enabled it gives attackers a useful view into kernel
+  memory.
+
+```
+  CONFIG_PROC_KCORE=n
+```
+
+### Disable KGDB
+
+  The Linux kernel supports KGDB over USB and console ports. These
+  mechanisms are controlled by the kgdbdbgp and kgdboc kernel
+  command-line parameters. It is important to ensure that no shipping
+  product contains a kernel with KGDB compiled-in.
+
+```
+  CONFIG_KGDB=n
+```
+
+### Disable Kprobes
+
+  Kprobes enables you to dynamically break into any kernel routine and
+  collect debugging and performance information non-disruptively. You
+  can trap at almost any kernel code address, specifying a handler
+  routine to be invoked when the breakpoint is hit.
+
+```
+  CONFIG_KPROBES=n
+```
+
+### Disable Tracing
+
+  FTrace enables the kernel to trace every kernel function. Providing kernel
+  trace functionality would assist an attacker in discovering attack vectors.
+
+```
+  CONFIG_FTRACE=n
+```
+
+### Disable Profiling
+
+  Profiling and OProfile enables profiling the whole system, include the kernel,
+  kernel modules, libraries, and applications.Providing profiling functionality
+  would assist an attacker in discovering attack vectors.
+
+```
+  CONFIG_OPROFILE=n
+  CONFIG_PROFILING=n
+```
+
+### Disable magic sysrq support
+
+  On a few architectures, you can access a powerful debugger interface
+  from the keyboard. The same powerful interface can be present on the
+  serial console – responding to serial break – of Linux on other
+  architectures. Disable to avoid potentially exposing this powerful
+  backdoor.
+
+```
+  CONFIG_MAGIC_SYSRQ=n
+```
+
+### Disable OOPS print on BUG()
+
+  The output from OOPS print can be helpful in Return Oriented
+  Programming (ROP) when trying to determine the effectives of an
+  exploit.
+
+```
+  CONFIG_DEBUG_BUGVERBOSE=n
+```
+
+### Disable kexec
+
+  This prevents someone who gets root from supplanting the kernel. This
+  can be used as a way to bypass signed kernels.
+
+```
+  CONFIG_KEXEC=n
+```
+
+### Disable kernel IP autoconfiguration
+
+  It is preferable to have IP configuration performed using a user-space
+  tool as these tend to have more validation. We do not want the network
+  interface coming up until the system has come up properly.
+
+```
+  CONFIG_IP_PNP=n
+```
+
+### Disable /proc/config.gz
+
+  It is extremely important to not expose the kernel configuration used
+  on a production device to a potential attacker. With access to the
+  kernel config, it could be possible for an attacker to build a custom
+  kernel for the device that may disable critical security features.
+
```

Comment:
I don't think the explanation is very good and should be improved (this is a tiny part but on the reason for an attacker to be able build a custom kernel). CONFIG_IKCONFIG disables the CONFIG within the kernel image iself.  /proc/config.gz can be disabled with CONFIG_IKCONFIG_PROC=N only. 

my 0,02€ ...